core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -197,12 +198,16 @@ public void beforeTest() {
         if (randomBoolean()) {
             QueryShardContext.setTypes(types);
         } else {
-            TestSearchContext testSearchContext = new TestSearchContext();
-            testSearchContext.setTypes(types);
-            SearchContext.setCurrent(testSearchContext);
+           setSearchContext(types); // TODO should this be set after we parsed and before we build the query? it makes more sense?
```

Comment:
makes sense, probably also the `QueryShardContext.setTypes(types);` should be postponed.